corehq/apps/reports/view_helpers.py

```diff
@@ -74,7 +74,8 @@ def _sortkey(child):
 
 
 def get_case_hierarchy(case):
-    def get_children(case, seen):
+    def get_children(case, seen=None):
+        seen = seen or set()
```

Comment:
I didn't notice this until re-review, and your original change would have fixed the issue, but I think this may have a bug in it. If the passed in set is empty a new set will be created in a recursive call, which may then add items that should but will not be seen by the outer call.
```suggestion
        if seen is None:
            seen = set()
```
~The way it was written before with a mutable default `def get_children(case, seen=set()):` is syntactically simpler and would work as well, although I'm guessing you think that smells.~ Well nevermind. It appears this bug was also in the original code. Not sure what made me think there was a mutable default.

Reply:
Oh good point, I hadn't noticed that.  I think I'll actually change it back to the way I had it before, which I find more explicit - we know exactly how this is going to be used, so why fork the logic?  I'd agree with you and Jenny  if this function were a utility or public interface, but it's an internal function used for a recursive loop.  I don't think the function declaration and invocation are separate in the same way normal functions should be.